test/python/test_reactor.py

```diff
@@ -1193,6 +1206,178 @@ def test_adaptive_precon_integration(self):
+        # conditions for simulation
+        yml = "h2o2.yaml"
+        # gas kinetics
+        gas = ct.Solution(yml, "ohmech")
+        gas.TPX = 1000, ct.one_atm, "H2:1.0, H:2.0, H2O:2.0"
+        gas.set_multiplier(0)
+        gas.set_multiplier(1, 14)
+        # create reactor
+        r = ct.IdealGasConstPressureMoleReactor(gas)
+        r.volume = 2
+        # create network
+        net = ct.ReactorNet([r])
+        net.step()
+        # the volume row is not considered in comparisons because it is presently
+        self.assertArrayNear(r.jacobian, r.finite_difference_jacobian, 1e-3, 1e-4)
+
+    def test_coverage_dependence_flags(self):
+        gas = ct.Solution("ptcombust.yaml", "gas")
+        surf = ct.Interface("ptcombust.yaml", "Pt_surf", [gas])
+        surf.TP = 900, ct.one_atm
+        surf.coverages = {"PT(S)":1}
+        with self.assertRaises(NotImplementedError):
+            surf.net_rates_of_progress_ddCi
+        # set skip and try to get jacobian again
+        surf.derivative_settings = {"skip-coverage-dependence": True}
+        surf.net_rates_of_progress_ddCi
+
+    def test_electrochemistry_flags(self):
+             # Phases
+        mech = "lithium_ion_battery.yaml"
+        anode, cathode, metal, electrolyte = ct.import_phases(
+            mech, ["anode", "cathode", "electron", "electrolyte"])
+        anode_int = ct.Interface(
+            mech, "edge_anode_electrolyte", adjacent=[anode, metal, electrolyte])
+        with self.assertRaises(NotImplementedError):
+            anode_int.net_rates_of_progress_ddCi
+        # set skip and try to get jacobian again
+        anode_int.derivative_settings = {"skip-electrochemistry": True}
+        anode_int.net_rates_of_progress_ddCi
+
+    def test_phase_order_surf_jacobian(self):
+        # create gas phase
+        gas_def = """
+        phases:
+        - name: gas
+          species:
+          - gri30.yaml/species: [H2, H, O, O2, OH, H2O, HO2, H2O2, CH3, CH4, CO, CO2,
+              HCO, CH2O, CH3O, CH3OH, N2, AR]
+          thermo: ideal-gas
+          kinetics: gas
+          reactions:
+          - gri30.yaml/reactions: declared-species
+          skip-undeclared-third-bodies: true
+        """
+        gas = ct.Solution(yaml=gas_def)
+        # set gas phase conditions
+        T0 = 1200
+        P0 = 25*ct.one_atm
+        X0 = 'CH4:0.5, H2O:0.2, CO:0.3'
+        gas.TPX = T0, P0, X0
+        # create reactors
+        r1 = ct.IdealGasMoleReactor(gas)
+        r2 = ct.IdealGasMoleReactor(gas)
+        r1.volume = 0.1
+        r2.volume = 0.1
+        # create solid and interfaces
+        solid = ct.Solution('diamond.yaml', 'diamond')
+        # first interface
+        interface1 = ct.Interface('diamond.yaml', 'diamond_100',
+                                    (gas, solid))
+        # interface with reversed phase order
+        interface2 = ct.Interface('diamond.yaml', 'diamond_100',
+                                    (solid, gas))
```

Comment:
Indentation here is a bit weird -- each of these would fit just fine on a single line.